source/customization/menus.rst

```diff
@@ -1,5 +1,367 @@
+and the menu item with the ID
+will be the first item in this section.
+All following items in the file
+will then be appended to the ID's section,
+until another item with an ID is specified.
+
+If two menu items
+from different ``.sublime-menu`` files
+reference the same item via ID,
+Sublime Text will override the previously defined item's parameters
+with the new parameters,
+if there are any.
+Child elements in a sub menu are appended
+and do not override.
+All following items are then appended to the ID's section,
+until another item with an ID is specified.
+
+Common practice is
+to assign IDs to separators
+and items having a sub-menu,
+to allow other packages or the user themselves
+to easily customize the menu.
+Separators always mark the beginning of a section,
+while it is required to address the exact same menu item
+in order to append items to its sub-menu.
+
+.. note::
+
+   Due to the strict forward lookup
+   it is possible to have
+   *multiple different items with the same ID*
+   in one menu.
+
+   Because of the potential confusion this may cause
+   it is discouraged.
+
+   Example:
+   The following three IDs
+   are defined in a menu, in order:
+   ``test, test2, test``.
+
+   The item with the second "test" ID
+   can then be targeted
+   using the following ID combinations:
+   ``test, test2, test``; ``test, test`` or ``test2, test``.
+
+
+Sub-Menus
+=========
+
+Every menu item can have a sub-menu,
+where hovering the item
+reveals the items grouped behind it.
+Sub-menus are independent menus
+with their own ID hierarchy.
+
+In order to extend a sub-menu
+from a different menu file,
+an ID must be specified in both occasions
+to target the correct item.
+
+
+The Main Menu
+=============
+
+Unlike all other menus,
+the Main Menu's root menu
+represents the menu items in the menu bar,
+for example "File" and "Help".
+Other than that,
+there is no notable difference.
+
+
+Interface For Commands
+======================
+
+Each menu item can be dynamically
+
+#. hidden,
+#. disabled,
+#. checked, or
+#. assigned a different caption.
+
+For this,
+commands may provide method endpoints in their class.
+The endpoints are called with the same arguments
+as the actual command would be
+(except for ``edit`` in ``TextCommand`` s),
+however they may also accept no parameters at all.
+
+#. ``is_visible``
+#. ``is_enabled``
+#. ``is_checked``
+#. ``description``
+
+Note that these are also relevant
+for the command palette
+to an extent.
+
+.. seealso::
+
+   `Official API Documentation on the Command endpoints`__
+
+   .. __: http://www.sublimetext.com/docs/3/api_reference.html#sublime_plugin.ApplicationCommand
+
+
+Context Menus in the Side Bar
+=============================
+
+The **Side Bar** and **Side Bar Mount Point** menus
+are different to the others
+in that they have **contextual information** available
+regarding the selected item(s).
+The selected items in the sidebar,
+both directories and files,

```

Comment:
Just leave this here and remove the above:

    The selected directories and files...